Documentation/gettingstarted/kubeproxy-free.rst

```diff
@@ -105,31 +108,179 @@ is ready to operate:
-the kube-proxy replacement instead of bailing out if the kernel support is
-missing. This makes the assumption that Cilium's BPF kube-proxy replacement would
-co-exist with kube-proxy on the system to optimize Kubernetes services. Given
-we've used kubeadm to deploy a kube-proxy-free setup, the ``strict`` mode
-has been used instead.
+underlying Linux kernel support is missing.
+
+Without explicitly specifying a ``kubeProxyReplacement`` option, helm uses
+``kubeProxyReplacement`` with ``probe`` by default which would automatically
+disable a subset of the features to implement the kube-proxy replacement instead
+of bailing out if the kernel support is missing. This makes the assumption that
+Cilium's BPF kube-proxy replacement would co-exist with kube-proxy on the system
+to optimize Kubernetes services. Given we've used kubeadm to explicitly deploy
+a kube-proxy-free setup, the ``strict`` mode has been used instead to ensure
+that we do not rely on a (non-existing) fallback.
+
+Cilium's BPF kube-proxy replacement is supported in direct routing as well as in
+tunneling mode.
+
+Direct Server Return (DSR) and other NodePort Settings
+######################################################
 
 When running Cilium's BPF kube-proxy replacement, by default, a NodePort or
 ExternalIPs service will be accessible through the IP address of a native device
 which has the default route on the host. To change the device, set its name in
 the ``global.nodePort.device`` helm option.
 
 In addition, thanks to the :ref:`host-services` feature, the NodePort service can
-be accessed from a host or a Pod within a cluster via it's public, cilium_host
-device or loopback address, e.g. ``127.0.0.1:NODE_PORT``.
+be accessed by default from a host or a Pod within a cluster via it's public,
+cilium_host device or loopback address, e.g. ``127.0.0.1:NODE_PORT``.
+
+By default, Cilium's BPF NodePort implementation operates in SNAT mode. That is,
+when node-external traffic arrives and the node determines that the backend for
+the NodePort or ExternalIPs service is at a remote node, then the node is redirecting
+the request to the remote backend on its behalf by performing SNAT. This does not
+require any additional MTU changes at the cost that replies from the backend need
+to make the extra hop back that node in order to perform the reverse SNAT translation
+there before returning the packet directly to the external client.
+
+This setting can be changed through the ``global.nodePort.mode`` helm option to
+``dsr`` in order to let Cilium's BPF NodePort implementation operate in DSR mode.
+In this mode, the backends reply directly to the external client without taking
+the extra hop, meaning, backends reply by using the service IP/port as a source.
+Another advantage is that while in the SNAT mode a client's source IP address is not
+preserved, in the DSR mode it is. Given a specific backend can be used by multiple
```

Comment:
This sentence would be more natural if it doesn't end with "it is", as this phrase usually bridges two parts of a sentence.

Suggestion:
```
Another advantage in DSR mode is that the client's source IP is preserved, so policy can match on it at the backend node. In the SNAT mode this is not possible.
```

The SNAT mode

Reply:
Makes sense, will change.